src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp

```diff
@@ -963,13 +963,113 @@ ov::runtime::Tensor generate(const
+
+                size_t b_offset = height * width * b / b_step * 3 / 2;
+                if (ColorFormat::i420 == format) {
+                    size_t u_index = b_offset + height * width + y * width / 2 + x;
+                    size_t v_index = u_index + height * width / 4;
+                    input_yuv[u_index] = u_val;
+                    input_yuv[v_index] = v_val;
+                } else {
+                    size_t uv_index = b_offset + height * width + y * width + x * 2;
+                    input_yuv[uv_index] = u_val;
+                    input_yuv[uv_index + 1] = v_val;
+                }
+                size_t y_index = b_offset + y * 2 * width + x * 2;
+                input_yuv[y_index] = y_val;
+                input_yuv[y_index + 1] = y_val;
+                input_yuv[y_index + width] = y_val;
+                input_yuv[y_index + width + 1] = y_val;
+            }
+        }
+    }
+    return input_yuv;
+}
+
+void fill_tensor(ov::Tensor& tensor, ColorFormat format) {
+    size_t full_height = tensor.get_shape()[1];
+    size_t full_width = tensor.get_shape()[2];
+    int b_dim = static_cast<int>(full_height * 2 / (3 * full_width));
+    ASSERT_GT(b_dim, 1) << "Image height is invalid for I420 Accuracy test";
+    ASSERT_EQ(255 % (b_dim - 1), 0) << "Image height is invalid for I420 Accuracy test";
+    int b_step = 255 / (b_dim - 1);
+    auto input_image = color_test_image(full_width, full_width, b_step, format);
+    auto data_ptr = static_cast<uint8_t*>(tensor.data());
+    for (size_t j = 0; j < input_image.size(); ++j) {
+        data_ptr[j] = input_image[j];
+    }
+}
+
+template<typename T>
+ov::runtime::Tensor generate(const
+                             std::shared_ptr<ov::op::v8::I420toRGB>& node,
+                             size_t port,
+                             const ov::element::Type& elemType,
+                             const ov::Shape& targetShape) {
+    ov::Tensor tensor(elemType, targetShape);
+    fill_tensor(tensor, ColorFormat::i420);
+    return tensor;
+}
+template<typename T>
+ov::runtime::Tensor generate(const
+                             std::shared_ptr<ov::op::v8::I420toBGR>& node,
+                             size_t port,
+                             const ov::element::Type& elemType,
+                             const ov::Shape& targetShape) {
+    ov::Tensor tensor(elemType, targetShape);
+    fill_tensor(tensor, ColorFormat::i420);
+    return tensor;
+}
+
+template<typename T>
+ov::runtime::Tensor generate(const
+                             std::shared_ptr<ov::op::v8::NV12toRGB>& node,
+                             size_t port,
+                             const ov::element::Type& elemType,
+                             const ov::Shape& targetShape) {
+    ov::Tensor tensor(elemType, targetShape);
+    fill_tensor(tensor, ColorFormat::nv12);
+    return tensor;
+}
+template<typename T>
+ov::runtime::Tensor generate(const
```

Comment:
overloaded comparation & generation functors should be outside of custom namespace

Reply:
done